funnel/models/event.py

```diff
@@ -125,6 +134,24 @@ class Participant(BaseMixin, db.Model):
     __table_args__ = (db.UniqueConstraint('proposal_space_id', 'email'),)
 
     @classmethod
+    def get(cls, space, email, participant_attrs={}, create=False):
+        participant = cls.query.filter_by(proposal_space=space, email=email).first()
+        if create and not participant:
+            participant = cls(
```

Comment:
One option is to add an `update` flag, so if a record is already present (and if `update` is set), the record will be updated with the dictionary that was passed.